test/com/xilinx/rapidwright/design/TestDesign.java

```diff
@@ -87,4 +89,26 @@ public void checkDcpRoundtripModuleInstAnchor(@TempDir Path tempDir) throws IOEx
         mi = design.getModuleInst("inst");
         Assertions.assertEquals(oldAnchor, mi.getAnchor().toString());
     }
+
+    @Test
+    @CheckOpenFiles
+    public void checkDcpRoundtripParallel(@TempDir Path tempDir) {
+        final Path filenameRead = RapidWrightDCP.getPath("optical-flow.dcp");
+        final Path filenameWrite = tempDir.resolve("testWrite.dcp");
+
+        Design before = Design.readCheckpoint(filenameRead);
+
+        try {
+            ParallelismTools.setParallel(true);
+            before.writeCheckpoint(filenameWrite);
+        } finally {
+            ParallelismTools.setParallel(false);
+        }
+
+        Design after = Design.readCheckpoint(filenameWrite);
+
+        Assertions.assertEquals(before.getNetlist(), after.getNetlist());
+        Assertions.assertEquals(before.getSiteInstMap(), after.getSiteInstMap());
+        Assertions.assertEquals(before.getNetMap(), after.getNetMap());
```

Comment:
Let me try this. I seem to recall this was problematic because it would perform an ordered compare of the two collections, whereas comparing their maps would one-liner work.